recipes/two_tables_design_template.md

```diff
@@ -0,0 +1,159 @@
+# Two Tables Design Recipe Template
+
+_Copy this recipe template to design and create two related database tables from a specification._
+
+## 1. Extract nouns from the user stories or specification
+
+```
+# EXAMPLE USER STORY:
+# (analyse only the relevant part - here the final line).
+
+STRAIGHT UP
+
+As a Maker
+So that I can let people know what I am doing  
+I want to post a message (peep) to chitter
+
+As a maker
+So that I can see what others are saying  
+I want to see all peeps in reverse chronological order
+
+As a Maker
+So that I can better appreciate the context of a peep
+I want to see the time at which it was made
+
+As a Maker
+So that I can post messages on Chitter as me
+I want to sign up for Chitter
+
+HARDER
+
+As a Maker
+So that only I can post messages on Chitter as me
+I want to log in to Chitter
+
+As a Maker
+So that I can avoid others posting messages on Chitter as me
+I want to log out of Chitter
+
+ADVANCED
+
+As a Maker
+So that I can stay constantly tapped in to the shouty box of Chitter
+I want to receive an email if I am tagged in a Peep
+```
+
+```
+Nouns:
+
+peep, peep order, time peep made, sign up, lon in, log out
+```
+
+## 2. Infer the Table Name and Columns
+
+Put the different nouns in this table. Replace the example with your own nouns.
+
+| Record                | Properties             |
+| --------------------- | ------------------     |
+| peep                  | message, date, user_id
+| user                  | name, email, password
+
+1. Name of the first table (always plural): `peeps` 
+
+    Column names: `message`, `date`, `user_id`
+
+2. Name of the second table (always plural): `users` 
+
+    Column names: `name`, `email`, `password`
+
+## 3. Decide the column types.
+
+[Here's a full documentation of PostgreSQL data types](https://www.postgresql.org/docs/current/datatype.html).
+
+Most of the time, you'll need either `text`, `int`, `bigint`, `numeric`, or `boolean`. If you're in doubt, do some research or ask your peers.
+
+Remember to **always** have the primary key `id` as a first column. Its type will always be `SERIAL`.
+
+```
+# EXAMPLE:
+
+Table: peeps
+id: SERIAL
+message: text
+date: date
```

Comment:
what is best practice; for the table to create a date or to create a date in the app.rb? I don't know.